packages/runtime/src/ctx.ts

```diff
@@ -215,8 +215,9 @@ export class Ctx implements Context {
 
 	getCallerApplicationID(): number {
 		let callerApplicationID = 0;
-		if (this.innerTxAppIDCallStack.length > 0) {
-			callerApplicationID = this.innerTxAppIDCallStack[this.innerTxAppIDCallStack.length - 1];
+		// the top value in the `innerTxAppIDCallStack` is the own appID not the caller appID
+		if (this.innerTxAppIDCallStack.length > 1) {
+			callerApplicationID = this.innerTxAppIDCallStack[this.innerTxAppIDCallStack.length - 2];
```

Comment:
Good job on this new PR. Though it seems that you are using my commit from https://github.com/scale-it/algo-builder/pull/798/files but not adding commit in that branch, nor adding me as a co-author. TBH it feels not very fair to me.

Reply:
Actually, I think my fix was also not so perfect, tho it gives the wanted result. 
And it looks correct to me what you said here in https://github.com/scale-it/algo-builder/pull/798#issuecomment-1268772011.

Reply:
Oh sorry for making this unfair to you! I am not really familiar with git but I will make add your commit to this branch and make you a co-author

Reply:
It's ok 😃 I already have the "contributor" chip. You can do it the way you like!